lib/phoenix/router/helpers.ex

```diff
@@ -176,20 +176,37 @@ defmodule Phoenix.Router.Helpers do
       defp to_param(true), do: "true"
       defp to_param(data), do: Phoenix.Param.to_param(data)
 
-      defp segments(segments, [], _reserved) do
+      defp segments!(segments, [], _reserved, _opts) do
         segments
       end
 
-      defp segments(segments, query, reserved) do
+      defp segments!(segments, query, reserved, _opts) when is_list(query) or is_map(query) do
         dict = for {k, v} <- query,
                not ((k = to_string(k)) in reserved),
                do: {k, v}
 
+
         case Conn.Query.encode dict, &to_param/1 do
           "" -> segments
           o  -> segments <> "?" <> o
         end
       end
+
+      defp segments!(segments, query, reserved, {helper, opts, var_length}) do
+        arity = var_length + 3
+        call_vars = if var_length > 0, do: Enum.map(1..(var_length), &(":arg#{&1}")), else: []
```

Comment:
Those things are not literal atoms, are they? So maybe it should be `arg#{&1}` instead of `:arg#{&1}`?